src/Integration.Vsix.UnitTests/Analysis/DisableRuleCommandTests.cs

```diff
@@ -246,152 +231,33 @@ public void Execute_NonCriticalErrorSuppressed()
         public void Execute_CriticalErrorNotSuppressed()
         {
             // Arrange
-            var errorList = new Mock<IErrorList>();
-            errorList.Setup(x => x.TableControl).Throws(new StackOverflowException("exception xxx"));
+            var errorList = Mock.Of<IErrorList>();
+            var errorListHelper = new Mock<IErrorListHelper>();
+            var ruleId = It.IsAny<SonarCompositeRuleId>();
+            errorListHelper.Setup(x => x.TryGetRuleIdFromSelectedRow(out ruleId)).Throws(new StackOverflowException("exception xxx"));
 
             var testLogger = new TestLogger();
             var mockUserSettingsProvider = new Mock<IUserSettingsProvider>();
             var solutionTracker = CreateSolutionTracker(SonarLintMode.Standalone);
 
-            var command = CreateDisableRuleMenuCommand(errorList.Object, mockUserSettingsProvider.Object, solutionTracker, testLogger);
+            var command = CreateDisableRuleMenuCommand(mockUserSettingsProvider.Object, solutionTracker, testLogger, errorListHelper.Object);
             Action act = () => command.Invoke();
 
-            // Act 
+            // Act
             act.Should().ThrowExactly<StackOverflowException>().And.Message.Should().Be("exception xxx");
 
             testLogger.AssertPartialOutputStringDoesNotExist("exception xxx");
         }
 
-        private static MenuCommand CreateDisableRuleMenuCommand(IErrorList errorList, IUserSettingsProvider userSettingsProvider, IActiveSolutionBoundTracker solutionTracker, ILogger logger)
+        private static MenuCommand CreateDisableRuleMenuCommand(IUserSettingsProvider userSettingsProvider, IActiveSolutionBoundTracker solutionTracker, ILogger logger, IErrorListHelper errorListHelper)
         {
             var dummyMenuService = new DummyMenuCommandService();
-            new DisableRuleCommand(dummyMenuService, errorList, userSettingsProvider, solutionTracker, logger);
+            new DisableRuleCommand(dummyMenuService, userSettingsProvider, solutionTracker, logger, errorListHelper);
 
             dummyMenuService.AddedMenuCommands.Count.Should().Be(1);
             return dummyMenuService.AddedMenuCommands[0];
         }
 
-        #region TryGetErrorCode tests
```

Comment:
Nice. The fact that the old class needed all of these logically separate tests suggests it was doing too much.